suite/suite.go

```diff
@@ -111,6 +114,9 @@ func Run(t *testing.T, suite TestingSuite) {
 		test := testing.InternalTest{
 			Name: method.Name,
 			F: func(t *testing.T) {
+				defer func() {
```

Comment:
Can we not just `defer testSync.Done()` instead of wrapping it in an anonymous function?

Reply:
Had some logic which i've removed later. good catch, have updated.